### DISCUSSION TITLE
Pricing regarding running a django server on codespaces

### DISCUSSION THREAD

**Discussion by anjanesh** [community/community#42121] [Codespaces] [answered]
### Select Topic Area

Question

### Body

If I wanted to use codespaces-django :

2 cores
60 hours free /month
$0.18 /additional hour

If manage.py runserver runs for a total of 400 hours in a month, then I'll be billed $61 ? Is this how billing is calculated ?

**Comment by craiglpeters** [comment, marked as answer]
If I understand your use case of running a service long-term, Codespaces is not likely the best fit. Codespaces is designed to be an ephemeral cloud-based dev environment. Aligned with this use, Codespaces has features to help users who have moved on to other tasks, and not shut down their dev environments, to [do so automatically](https://docs.github.com/en/codespaces/customizing-your-codespace/setting-your-timeout-period-for-github-codespaces) so they don't use up all their free usage on an idle machine.

Additionally Codespaces also includes monthly free usage for storage, and bills for storage beyond that free usage. You may find [this post](https://github.com/community/community/discussions/39697) helpful to better understand the billing.
